IntelliTect.TestTools.TestFramework/IntelliTect.TestTools.TestFramework/TestBuilder.cs

```diff
@@ -79,6 +79,12 @@ public TestBuilder AddDependencyService<T>()
             return this;
         }
 
+        public TestBuilder AddDependencyService<TServiceType,TImplementationType>()
+        {
+            Services.AddScoped(typeof(TServiceType), typeof(TImplementationType));
```

Comment:
Why does this add a Scoped instance while the other `AddDependencyService` methods do singleton?

Reply:
Ah, okay, so it depends on how the item is being added. If it's by type, the assumption (and so far, the need to-date) is that it's something you want managed solely by the DI service within the context of a single test case (hence, scoped). If it's added by object, then it's added a singleton on the assumption that something *outside* the scope of the DI container is managing that object (e.g., xUnit's data collections in order to share an object amongst multiple tests, which xUnit will handle cleaning up.)